docs/CODEBASE_OVERVIEW.md

```diff
@@ -0,0 +1,287 @@
+
+## Service API
+**`service` Gradle sub-project**
+
+Below is an outline of the project structure. More details in the subsections below.
+```
+local-dev               # Utilities for local development
+bio.terra.tanagra
+    app                 # API entry points and application configuration
+    db                  # Application database
+    service
+      accesscontrol     # Access control
+      artifact          # API artifacts
+      authentication    # Authentication
+      export            # Data export
+```
+
+### Utilities for local development
+**`service` Gradle sub-project `local-dev` directory**
+
+Helper scripts for running the service locally live in this directory.
+
+### API entry points and application configuration
+**`service` Gradle sub-project `app` package**
+
+Classes related to OpenAPI and SpringBoot web framework live in this package.
+- All application/deployment configuration properties are defined in the `configuration` package. Each new property 
+should include an annotation that will be used to generate documentation.
+- All API endpoints map to a method in one of the `controller` package classes. The endpoints are defined in the
+OpenAPI specification file in the `resources/api` directory.
+
+### Application database
+**`service` Gradle sub-project `db` package**
+
+The application database stores the service state (e.g. a user's cohorts). We support PostGres, MariaDB, and MySQL.
+We use Liquibase to manage the database schema; the changeset files live in the `resources/db` directory.
+
+### Access control
+**`service` Gradle sub-project `accesscontrol` package**
+
+Access control/authorization is configurable per deployment. The various access control implementations, and the 
+interface they share live in this package. More information in [this readme](./ACCESS_CONTROL.md).
+
+### API artifacts
+**`service` Gradle sub-project `artifact` package**
+
+Artifacts that are exposed through the API include studies, cohorts, reviews, concept sets, and activity logs. These
+artifacts are part of the application persistent state and stored in the application database.
+
+When adding a new artifact or logic to an existing one, prefer defining a POJO-like model class and putting logic into
+the associated service or dao classes.
+
+### Authentication
+**`service` Gradle sub-project `authentication` package**
+
+We support three types of authentication tokens: bearer access token, JWT from IAP in front of GKE, JWT from IAP in
+front of AppEngine. Utilities for processing these tokens lives in this package.
+
+### Data export
+**`service` Gradle sub-project `export` package**
+
+Data export models are configurable per deployment. The various data export implementations, and the interface they
+share live in this package. More information in [this readme](./DATA_EXPORT.md).
+
+## UI
+**`ui` top-level directory**
+
+Web UI code. More information in [this readme](../ui/README.md).
```

Comment:
Would it be better to have a `UI.md` here instead keep all of the docs in the same place?

Reply:
OK with me. I'll move the current `ui/README.md` here then.

Reply:
Done. Also added a note there about opening Cypress, which I keep forgetting.